setup

```diff
@@ -69,14 +93,55 @@ install() {
     cp "${INSTALLDIR}/containers.conf" "${ROOTFS}/etc/containers/"
     cp "${INSTALLDIR}/contexts" "${ROOTFS}/usr/share/containers/selinux/"
     cp "${INSTALLDIR}/file_contexts" "${ROOTFS}/etc/selinux/targeted/contexts/files/file_contexts"
-    replaceIDs "${ROOTFS}/etc/subuid" containers ${qmContainerIDs}
-    replaceIDs "${ROOTFS}/etc/subgid" containers ${qmContainerIDs}
+    replaceIDs "${ROOTFS}/etc/subuid" containers ${QM_CONTAINER_IDS}
+    replaceIDs "${ROOTFS}/etc/subgid" containers ${QM_CONTAINER_IDS}
     hirteSetup "${ROOTFS}"
     unshare --mount-proc -R "${ROOTFS}" -m systemctl enable hirte-agent.service
     storage "${ROOTFS}"
     restorecon -R "${ROOTFS}"
 }
 
+# read command line arguments
+opts=$(getopt \
+  --longoptions "$(printf "help,%s:," "${CMDLINE_ARGUMENT_LIST[@]}")" \
+  --name "$(basename "$0")" \
+  --options "" \
+  -- "$@"
+)
+
+eval set --"${opts}"
+
+# main()
+root_check
+
+echo "setup has started..."
+echo  "  * rootfs: ${ROOTFS}"
```

Comment:
Shouldn't this happen after the getopt loop below?

Reply:
In one of the patches below, I changed that. Sorry,if you prefer I can do a squash.